project/src/UI/Upgrades/UpgradeButton.gd

```diff
@@ -3,13 +3,35 @@
 tool
 extends TextureButton
 
+signal on_hide_complete;
+signal on_show_complete;
+
 export var texture: Texture setget set_texture
 export var text := "" setget set_text
 
 onready var texture_rect := $VBoxContainer/TextureRect
 onready var label := $VBoxContainer/Label
+onready var animation_player := $AnimationPlayer
+
+
+func show_delayed(delay : float = 0) -> void:
```

Comment:
You can also use public functions that are coroutines directly from the caller to keep your code linear. We name these functions `*_async` when they're intended to be used as coroutines.